hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/server/TopicMetadataLoadingRunner.java

```diff
@@ -1,81 +1,154 @@
+    public TopicMetadataLoadingRunner(BrokerTopicMetadataFetcher brokerTopicMetadataFetcher,
+                                      String localDatacenter,
+                                      List<String> datacenters,
+                                      TopicsCache topicsCache,
+                                      int retryCount,
+                                      Duration retryInterval,
+                                      int threadPoolSize) {
+        this.brokerTopicMetadataFetcher = brokerTopicMetadataFetcher;
+        this.localDatacenter = localDatacenter;
+        this.datacenters = datacenters;
+        this.topicsCache = topicsCache;
+        ThreadFactory threadFactory = new ThreadFactoryBuilder().setNameFormat("topic-metadata-loader-%d").build();
+        this.scheduler = Executors.newScheduledThreadPool(threadPoolSize, threadFactory);
+        this.retryPolicy = new RetryPolicy<MetadataLoadingResult>()
+                .withMaxRetries(retryCount)
+                .withDelay(retryInterval)
+                .handleIf((resp, cause) -> resp.isFailure());
+    }
 
-    private final int threadPoolSize;
+    public boolean refreshMetadataForLocalDatacenter() {
+        return refreshMetadataForDatacenters(List.of(localDatacenter));
+    }
 
-    public TopicMetadataLoadingRunner(BrokerMessageProducer brokerMessageProducer,
-                               TopicsCache topicsCache,
-                               int retryCount,
-                               Duration retryInterval,
-                               int threadPoolSize) {
-        this.brokerMessageProducer = brokerMessageProducer;
-        this.topicsCache = topicsCache;
-        this.retryCount = retryCount;
-        this.retryInterval = retryInterval;
-        this.threadPoolSize = threadPoolSize;
+    public void refreshMetadataForAllDatacenters() {
+        refreshMetadataForDatacenters(datacenters);
     }
 
-    public List<MetadataLoadingResult> refreshMetadata() throws Exception {
-        long start = System.currentTimeMillis();
-        logger.info("Loading topics metadata");
-        List<CachedTopic> topics = topicsCache.getTopics();
-        List<MetadataLoadingResult> allResults = loadMetadataForTopics(topics);
-        logResultInfo(allResults, System.currentTimeMillis() - start);
-        return allResults;
+    private boolean refreshMetadataForDatacenters(List<String> datacenters) {
+        try {
+            long start = System.currentTimeMillis();
+            logger.info("Loading topic metadata for datacenters: {}", datacenters);
+            List<CachedTopic> topics = topicsCache.getTopics();
+            List<MetadataLoadingResult> allResults = loadMetadataForTopics(topics, datacenters);
+            logger.info("Finished loading topic metadata in {}ms", System.currentTimeMillis() - start);
+            logResultInfo(allResults);
+            return allResults.stream().anyMatch(MetadataLoadingResult::isFailure);
```

Comment:
`allResults.stream().anyMatch(MetadataLoadingResult::isFailure)` means that we return `true` if any job fails. Shouldn't we negate the result?

Reply:
`KafkaReadinessCheckTest` is probably failing because of this

Reply:
Correct. Fixed.